src/Symfony/Component/DependencyInjection/Compiler/MergeExtensionConfigurationPass.php

```diff
@@ -59,6 +66,43 @@ public function process(ContainerBuilder $container)
                 $tmpContainer->addObjectResource($configuration);
             }
 
+            if (null !== $envTypes && method_exists(BaseNode::class, 'setPlaceholder')) {
+                $defaultEnvBag = new ParameterBag($parameters);
+                foreach ($resolvingBag->getEnvPlaceholders() as $env => $placeholders) {
+                    $prefix = (false === $i = strpos($env, ':')) ? 'string' : substr($env, 0, $i);
+                    $types = isset($envTypes[$prefix]) ? $envTypes[$prefix] : array('string');
+                    $values = array();
+                    $default = null;
+                    if ($hasEnv = (false === $i && $defaultEnvBag->has("env($env)"))) {
+                        switch ($type = gettype($default = $defaultEnvBag->get("env($env)"))) {
+                            case 'boolean': $type = 'bool'; break;
+                            case 'double': $type = 'float'; break;
+                            case 'integer': $type = 'int'; break;
+                        }
+                        if (null !== $default && !in_array($type, $types, true)) {
+                            throw new LogicException(sprintf('Invalid type for env parameter "env(%s)", expected "%s" got "%s".', $env, implode('", "', $types), $type));
+                        }
+                    }
+                    foreach ($types as $type) {
+                        if ($hasEnv) {
+                            $values[$type] = $default;
+                            continue;
+                        }
+                        switch ($type) {
+                            case 'array': $values[$type] = array(); break;
+                            case 'bool': $values[$type] = false; break;
+                            case 'float': $values[$type] = 0.0; break;
+                            case 'int': $values[$type] = 0; break;
+                            case 'string': $values[$type] = ''; break;
+                            default: $values[$type] = null; break;
```

Comment:
is this case possible? if not, default should be string, isn't it?

Reply:
only if we forget the update it after adding types. Should be never, just a safety net.

`default: throw` then? In case of non-fatal behavior id say `null`.. why assume string?

Reply:
no default case and let php crash? fine fore me :)